product-service/src/handlers/get-product-by-id.js

```diff
@@ -0,0 +1,34 @@
+const axios = require('axios');
+
+const productList = require('../db.json');
+
+export const getProductById = async (event) => {
+  try {
+    const { productId } = event.pathParameters;
+    const product = productList.find(item => item.id === productId);
+
+    // just a random request for making the task with async/await
+    await axios.get('https://jsonplaceholder.typicode.com/todos/1');
+
+    if (!product) throw new Error();
```

Comment:
Why? Why not just return a 404 response? Furthermore, you will always get 404 if an error is thrown at another level, e.g. in DB.

Reply:
yes, it could be done better :)